src/opcodes/tests/test_opcodes.cairo

```diff
@@ -70,6 +70,25 @@ fn test_op_max() {
     assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
 }
 
+#[test]
+fn test_op_within() {
+    let program = "OP_1 OP_0 OP_1 OP_WITHIN";
+    let mut compiler = CompilerTraitImpl::new();
+    let bytecode = compiler.compile(program);
+    let mut engine = EngineTraitImpl::new(bytecode);
+    let _ = engine.step();
+    let _ = engine.step();
+    let _ = engine.step();
+    let res = engine.step();
+    assert!(res, "Execution of run failed");
+
+    let dstack = engine.get_dstack();
+    assert_eq!(dstack.len(), 1, "Stack length is not 1");
+
+    let expected_stack = array!["\0\0\0\0\0\0\0\0"];
+    assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
+}
```

Comment:
Nice works, maybe you can add one more test that returns 1. You can also use opcode like OP_2 OP_3 etc...

Reply:
Okay will do it now

Reply:
Hi @j1mbo64 I just added a second test for when 1 is returned and I used OP_3, OP_0 and OP_1 in my tests